test/functional/README.md

```diff
@@ -0,0 +1,376 @@
+expectation as an array, with one object for each measurement. Then, the suite will recognize the array length and will
+expect the same number of NGSI requests. I.E:
+
+```js
+[
+    {
+        id: 'TheLightType2:MQTT_2',
+        type: 'TheLightType2',
+        temperature: {
+            value: 10,
+            type: 'Number'
+        },
+        status: {
+            value: false,
+            type: 'Boolean'
+        }
+    },
+    {
+        id: 'TheLightType2:MQTT_2',
+        type: 'TheLightType2',
+        temperature: {
+            value: 20,
+            type: 'Number'
+        },
+        status: {
+            value: true,
+            type: 'Boolean'
+        }
+    }
+];
+```
+
+You also should define the measure as multimeasure. This is done by defining the measure JSON element as an array of
+objects. Each object will be a measure that will be sent to the Context Broker in a different request. I.E:
+
+```javascript
+measure: {
+    url: 'http://localhost:' + config.http.port + '/iot/json',
+    method: 'POST',
+    qs: {
+        i: 'MQTT_2',
+        k: '123456'
+    },
+    json: [
+        {
+            s: false,
+            t: 10
+        },
+        {
+            s: true,
+            t: 20
+        }
+    ]
+}
+```
+
+#### Transport
+
+The test suite supports using the internal node lib function `iotAgentLib.update`, `HTTP` or `MQTT` for measure sending.
+In order to select the specific way to send the measure you should add a `transport` element to each should case having
+the value set to `MQTT`. By doing so, the suite will automatically configure the mock to connect to the MQTT broker and
+send the measure to the correct topic based on the `i` and `k` parameters. It will ignore the `url` and `method`
+parameters present in the measure JSON element. I.E:
+
+```javascript
+should: [
+    {
+        transport: 'MQTT',
+        shouldName: 'should send its value to the Context Broker when using MQTT',
+        type: 'single',
+        measure: {
+            url: 'http://localhost:' + config.http.port + '/iot/json',
+            method: 'POST',
+            qs: {
+                i: 'MQTT_2',
+                k: '123456'
+            },
+            json: {
+                s: false,
+                t: 10
+            }
+        },
+        expectation: {
+            id: 'TheLightType2:MQTT_2',
+            type: 'TheLightType2',
+            temperature: {
+                value: 10,
+                type: 'Number'
+            },
+            status: {
+                value: false,
+                type: 'Boolean'
+            }
+        }
+    }
+];
+```
+
+#### No payload reception
+
+The test suite also supports the case in which the Context Broker does not receive any payload. This is done by defining
+the expectation as an empty object. I.E:
+
+```javascript
+    ...
+    expectation: []
+    ...
+```
+
```

Comment:
```suggestion

Note that this means the CB *must not* receive any payload. In other words, if the CB would receive any payload, the test will fail.

```

To clarify that `[]` here means "we check nothing is recieved" versus "no check is done".